File: SYP.Web/Modules/ServerTypes/Setting/VendorTypeEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { VendorTypeRow } from "./VendorTypeRow";

export interface VendorTypeEditorColumns {
    Id: Column<VendorTypeRow>;
    Title: Column<VendorTypeRow>;
    DiscountType: Column<VendorTypeRow>;
    DiscountValue: Column<VendorTypeRow>;
    Description: Column<VendorTypeRow>;
    IsActive: Column<VendorTypeRow>;
}

export class VendorTypeEditorColumns extends ColumnsBase<VendorTypeRow> {
    static readonly columnsKey = 'Setting.VendorTypeEditor';
    static readonly Fields = fieldsProxy<VendorTypeEditorColumns>();
}

File: SYP.Web/Modules/Setting/VendorType/Editor/VendorTypeEditorColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.VendorTypeEditor")]
[BasedOnRow(typeof(VendorTypeRow), CheckNames = true)]
public class VendorTypeEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string DiscountType { get; set; }
    public decimal DiscountValue { get; set; }
    public string Description { get; set; }
    public bool IsActive { get; set; }
}